src/overlays/actors/ovl_En_Stop_heishi/z_en_stop_heishi.h

```diff
@@ -2,16 +2,46 @@
 #define Z_EN_STOP_HEISHI_H
 
 #include "global.h"
+#include "objects/object_sdn/object_sdn.h"
 
 struct EnStopheishi;
 
 typedef void (*EnStopheishiActionFunc)(struct EnStopheishi*, PlayState*);
 
+#define ENSTOPHEISHI_GET_SWITCH_FLAG(this) (((this)->actor.params) & 0x3F)
+#define ENSTOPHEISHI_GET_C(this) ((((this)->actor.params) >> 0xC) & 0xF)
```

Comment:
```suggestion
#define ENSTOPHEISHI_GET_F000(this) ((((this)->actor.params) >> 0xC) & 0xF)
```
It grabs the `0xF000` bits from `params`. Also what issues were you having with using `thisx` again?

Reply:
Changing the macros to take `thisx`, and than just using `&this->actor` in `Init` works fine for me.

Reply:
Weird, all of the stop_heishi spawned by the scenes themselves have 0x0 in that slot, so unused?

Reply:
Remember to rename this to `ENSTOPHEISHI_GET_F000`